pkg/client/interface.go

```diff
@@ -9,6 +9,29 @@ import (
 	"pocket/pkg/observable"
 )
 
+type BlockClient interface {
+	// Blocks returns an observable which emits newly committed blocks.
+	CommittedBlocksSequence(context.Context) BlocksObservable
+	//CommittedBlocksSequence() observable.Observable[Block]
+	// LatestBlock returns the latest block that has been committed.
+	LatestBlock(context.Context) Block
+	// Close unsubscribes all observers of the committed blocks sequence observable
+	// and closes the events query client.
+	Close()
+}
+
+// BlocksObservable is an observable which is notified with an either
+// value which contains either an error or the event message bytes.
+// TODO_HACK: The purpose of this type is to work around gomock's lack of
```

Comment:
Fwiw, I kind of like the explicit naming. I would even move it to the top of the file

Reply:
I'll clarity in the comment. The crux of the issue is that we can't make it an alias (i.e. `TypeB = TypeA`) because that is insufficient to work around the gomock incompatibility; however, not using an alias means that we don't  have access to `TypeA`'s methods from `TypeB`. This means that we would either have to cast the type everywhere we need to call a `TypeA` method or implement those methods on `TypeB` which would basically be a thin wrapper that does the same type cast and proxy's the method call.

Reply:
Again, I feel like we're doing the right thing here and that a thin wrapper that includes type casting is actually the correct & idiomatic approach, and not a hack :)

https://yourbasic.org/golang/type-alias/

![Screenshot 2023-10-26 at 10 45 10 AM](https://github.com/pokt-network/poktroll/assets/1892194/3578c665-154a-4502-a48f-5252492f7c60)

Reply:
I think I was conflating a generics issue with this. Thanks for helping clear that up. :sweat_smile: